### PR TITLE
Numismatic References and Citations

### LINKED ISSUE #2115: Nested numismatic references
Coin and NumismaticIssue references are difficult to collapse into a single string. We will probably need to create a References model and nest that in the coin and issue forms. Perhaps with data tables?

Child of #2025

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #2115

### CODE REVIEW

File: app/controllers/numismatic_citations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  self.change_set_class = DynamicChangeSet
+  self.resource_class = NumismaticCitation
+  self.change_set_persister = ::ChangeSetPersister.new(
+    metadata_adapter: Valkyrie::MetadataAdapter.find(:indexing_persister),
+    storage_adapter: Valkyrie.config.storage_adapter
+  )
+  before_action :load_numismatic_references, only: [:new, :edit]
+
+  def create
+    # Save append_id and then remove from resource params
```
Comment (hackmastera) [inline app/controllers/numismatic_citations_controller.rb]: can you add why to this comment?

Comment (hackmastera) [reply app/controllers/numismatic_citations_controller.rb]: you're using the append_id list to get the parents to attach to?

Comment (eliotjordan) [reply app/controllers/numismatic_citations_controller.rb]: Yes. The append_id param is used to retrieve the parent numismatic issue or coin. The citation id is then inserted into the parent `numismatic_citation_ids` array.

**Comment by eliotjordan** [comment]
@escowles I fixed an issue with one of the new model comments and expanded on the controller comment a little.

**Comment by eliotjordan** [comment]
@hackmastera This is ready for another review.

- Updated to not overload the `append_id` parameter and moved logic to the change set persister.
- Rebased against recent Figgy file structure re-org.

**Pull request merged**